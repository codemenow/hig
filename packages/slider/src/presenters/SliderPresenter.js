import React, { Component } from "react";
import PropTypes from "prop-types";
import { css } from "emotion";
import ThemeContext from "@hig/theme-context";
import stylesheet from "./SliderPresenter.stylesheet";

export default class SliderPresenter extends Component {
  static propTypes = {
    disabled: PropTypes.bool,
    hasFocus: PropTypes.bool,
    hasHover: PropTypes.bool,
    isPressed: PropTypes.bool,
    id: PropTypes.string,
    max: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    min: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    name: PropTypes.string,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onInput: PropTypes.func,
    onMouseDown: PropTypes.func,
    onMouseEnter: PropTypes.func,
    onMouseLeave: PropTypes.func,
    onMouseUp: PropTypes.func,
    step: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
  };

  render() {
    const {
      disabled,
      hasFocus,
      hasHover,
      isPressed,
      max,
      min,
      value,
      ...otherProps
    } = this.props;

    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles }) => {
          const rangeRange = max - min;
          const valueRatio = (value - min) / rangeRange;

          const styles = stylesheet(valueRatio, resolvedRoles);

          return (
            <input
              className={css(styles.input)}
              disabled={disabled}
              type="range"
              aria-valuemin={min}
              aria-valuemax={max}
              aria-valuenow={value}
              max={max}
              min={min}
              value={value}
              {...otherProps}
            />
          );
        }}
      </ThemeContext.Consumer>
    );
  }
}
