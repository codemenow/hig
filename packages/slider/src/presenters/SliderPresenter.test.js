import { takeSnapshotsOf } from "@hig/jest-preset/helpers";
import SliderPresenter from "./SliderPresenter";

describe("Slider/presenters/SliderPresenter", () => {
  takeSnapshotsOf(SliderPresenter, [{ desc: "renders", props: {} }]);
});
