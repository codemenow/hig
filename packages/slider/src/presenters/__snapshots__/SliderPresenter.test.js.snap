// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider/presenters/SliderPresenter renders 1`] = `
.emotion-0 {
  -webkit-appearance: none;
  width: 100%;
  background-color: transparent;
  box-sizing: content-box;
  height: 20px;
  margin: 0;
  position: relative;
  outline: none;
}

.emotion-0::-ms-tooltip {
  display: none;
}

.emotion-0::-ms-tooltip {
  display: none;
}

.emotion-0::-moz-focus-outer {
  border: 0;
}

.emotion-0::-ms-thumb {
  box-sizing: content-box;
  height: 16px;
  width: 16px;
  background-color: #D4DBE1;
  border: 0;
  border-radius: 50%;
  box-shadow: 0 0 0 2px #ffffff;
  cursor: pointer;
  position: relative;
  top: 50%;
  -webkit-appearance: none;
}

.emotion-0::-ms-thumb:hover {
  background-color: #0ED3BE;
  box-shadow: 0 0 0 2px rgba(255,255,255,0.15);
}

.emotion-0::-ms-thumb:focus {
  outline: none;
  background-color: #0696D7;
  box-shadow: 0 0 0 2px rgba(255,255,255,0.25);
}

.emotion-0::-ms-thumb:active {
  background-color: #0671A1;
  box-shadow: 0 0 0 4px rgba(255,255,255,0.25);
}

.emotion-0::-ms-thumb[disabled] {
  opacity: 0.3;
}

.emotion-0::-moz-range-thumb {
  box-sizing: content-box;
  height: 16px;
  width: 16px;
  background-color: #D4DBE1;
  border: 0;
  border-radius: 50%;
  box-shadow: 0 0 0 2px #ffffff;
  cursor: pointer;
  position: relative;
  top: 50%;
  -webkit-appearance: none;
}

.emotion-0::-moz-range-thumb:hover {
  background-color: #0ED3BE;
  box-shadow: 0 0 0 2px rgba(255,255,255,0.15);
}

.emotion-0::-moz-range-thumb:focus {
  outline: none;
  background-color: #0696D7;
  box-shadow: 0 0 0 2px rgba(255,255,255,0.25);
}

.emotion-0::-moz-range-thumb:active {
  background-color: #0671A1;
  box-shadow: 0 0 0 4px rgba(255,255,255,0.25);
}

.emotion-0::-moz-range-thumb[disabled] {
  opacity: 0.3;
}

.emotion-0::-webkit-slider-thumb {
  box-sizing: content-box;
  height: 16px;
  width: 16px;
  background-color: #D4DBE1;
  border: 0;
  border-radius: 50%;
  box-shadow: 0 0 0 2px #ffffff;
  cursor: pointer;
  position: relative;
  top: 50%;
  -webkit-appearance: none;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-0::-webkit-slider-thumb:hover {
  background-color: #0ED3BE;
  box-shadow: 0 0 0 2px rgba(255,255,255,0.15);
}

.emotion-0::-webkit-slider-thumb:focus {
  outline: none;
  background-color: #0696D7;
  box-shadow: 0 0 0 2px rgba(255,255,255,0.25);
}

.emotion-0::-webkit-slider-thumb:active {
  background-color: #0671A1;
  box-shadow: 0 0 0 4px rgba(255,255,255,0.25);
}

.emotion-0::-webkit-slider-thumb[disabled] {
  opacity: 0.3;
}

.emotion-0::-ms-track {
  width: 100%;
  height: 2px;
  border: none;
  background-color: rgba(212,219,225,0.5);
  cursor: pointer;
  color: transparent;
  outline: none;
}

.emotion-0::-ms-track[disabled] {
  opacity: 0.3;
}

.emotion-0::-moz-range-track {
  width: 100%;
  height: 2px;
  border: none;
  background-color: rgba(212,219,225,0.5);
  cursor: pointer;
  color: transparent;
  outline: none;
}

.emotion-0::-moz-range-track[disabled] {
  opacity: 0.3;
}

.emotion-0::-webkit-slider-runnable-track {
  width: 100%;
  height: 2px;
  border: none;
  background-color: rgba(212,219,225,0.5);
  cursor: pointer;
  color: transparent;
  outline: none;
  background-image: linear-gradient(#D4DBE1,#D4DBE1);
  background-position: 0;
  background-size: calc((0.5 * 16px)  + (NaN * (100% - 16px))) 100%;
  background-repeat: no-repeat;
}

.emotion-0::-webkit-slider-runnable-track[disabled] {
  opacity: 0.3;
}

.emotion-0::-ms-fill-lower {
  border: none;
  background-color: #D4DBE1;
}

.emotion-0::-moz-range-progress {
  border: none;
  background-color: #D4DBE1;
}

<input
  aria-valuemax={undefined}
  aria-valuemin={undefined}
  aria-valuenow={undefined}
  className="emotion-0"
  disabled={undefined}
  max={undefined}
  min={undefined}
  type="range"
  value={undefined}
/>
`;
